File: app/opensearch/query_builder.py
from dataclasses import dataclass, field


@dataclass
class QueryRequest:
    keyword: str = ""
    vector: list[float] = field(default_factory=list)
    filters: dict = field(default_factory=dict)
    size: int = 10


def normalize_request(req: QueryRequest) -> QueryRequest:
    if req.size <= 0:
        req.size = 10
    return req

File: backend/opensearch/query_builder.go
package opensearch

type QueryRequest struct {
	Keyword string
	Vector  []float32
	Filters map[string]any
	Size    int
}

func NormalizeRequest(req QueryRequest) QueryRequest {
	if req.Size <= 0 {
		req.Size = 10
	}
	if req.Filters == nil {
		req.Filters = map[string]any{}
	}
	return req
}
